src/networking.c

```diff
@@ -911,6 +911,13 @@ void unlinkClient(client *c) {
             c->client_list_node = NULL;
         }
 
+        /* In the case of diskless replication the fork is writing to the
+         * sockets and just closing the fd isn't enough, if we don't also
+         * shutdown the socket the fork will continue to write to the slave
+         * and the salve will only find out that it was disconnected when
+         * it will finish reading the rdb. */
+        shutdown(c->fd, SHUT_RDWR);
```

Comment:
Hi Oran, what about doing this call only if the other entry is a slave? Or at this stage it is yet not flagged as a slave? Any other way for us to understand this? Thanks.